src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -252,9 +287,13 @@ impl<'a, T: Timestamp + Lattice> StorageController<'a, T> {
     }
 }
 
-impl<'a, C: Client<T>, T: Timestamp + Lattice> StorageControllerMut<'a, C, T> {
-    /// Acquire a mutable reference to the collection state, should it exist.
-    pub(super) fn collection_mut(
+impl<'a, C: Client<T>, T> StorageControllerMut<'a, C, T>
+where
+    T: Timestamp + Lattice + TryInto<i64> + TryFrom<i64>,
+    <T as TryInto<i64>>::Error: std::fmt::Debug,
+    <T as TryFrom<i64>>::Error: std::fmt::Debug,
```

Comment:
These constraints seem to be everywhere, and it would be great to localize them instead! For example, having them only on the `persist_timestamp_bindings` method, or better yet not at all and having the stash trait link them up only at the constructor time.